internal/dcache/rpc/server/handler.go

```diff
@@ -756,6 +804,20 @@ func (mv *mvInfo) refreshFromClustermap() error {
 	for rvName, rvState := range newRVs {
 		common.Assert(cm.IsValidComponentRVState(rvState), rvName, mv.mvName, rvState)
 
+		//
+		// degrade-mv workflow marks component RVs as offline, for the RVs which are marked offline,
+		// but it doesn't commit the changes to clustermap before it runs the fix-mv workflow, the
+		// following achieves that w/o a clustermap update being forced after degrade-mv.
+		// Note that this is one of those "safe deductions" that we can do while taking the risk of
+		// deviating away from the actual clustermap, but note that clustermap will soon be updated
+		// to reflect it.
+		//
+		if cm.GetRVState(rvName) == dcache.StateOffline && rvState != dcache.StateOffline {
+			log.Warn("mvInfo::refreshFromClustermap: %s/%s state is %s while RV state is offline, marking component RV state as offline",
+				rvName, mv.mvName, rvState)
+			rvState = dcache.StateOffline
+		}
```

Comment:
When can we get this scenario?
We get rvState from newRVs list which we get from `cm.GetRVs(mv.mvName)`. So, `cm.GetRVState(rvName)` should be same as the rvState.